File: src/data/Model/Cms360/Admission/Students.cs
/*
*   Auther: H.Muhammad Kamran
*   email: hmuhdkamran@gmail.com
*   contact: +92 (313 / 333) 9112 845
*/

using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Cms360.Data.Model
{
    [Table("Students", Schema = "Admission")]
    public partial class AdmissionStudents
    {
        [Key]
        [Required]
        public Guid StudentId { get; set; }
        [Required]
        public string FullName { get; set; }
        [Required]
        public string FatherName { get; set; }
        [Required]
        public string StudentCNIC { get; set; }
        [Required]
        public string ParentCNIC { get; set; }
        [Column(TypeName = "jsonb")]
        public string StudentContactNo { get; set; }
        [Column(TypeName = "jsonb")]
        public string ParentContactNo { get; set; }
        [Column(TypeName = "jsonb")]
        public string Guardians { get; set; }
        [Required]
        public Guid GenderId { get; set; }
        [Required]
        public DateTime DateOfBirth { get; set; }
        [Column(TypeName = "jsonb")]
        public string Address { get; set; }
        [Column(TypeName = "jsonb")]
        public string AcademicInfo { get; set; }
        public Guid BloodGroupId { get; set; }
        public Guid ReligionId { get; set; }
        [Required]
        public int StatusId { get; set; }
        [Required]
        public Guid LoggerId { get; set; }
        public string Image { get; set; }
         

         [Column(TypeName = "jsonb")]
        public string Operation  {get;set;}

    }




    [Table("VWStudentRecord", Schema = "Admission")]
    public partial class StudentRecordVM
    {
        [Key]
        [Required]
        public Guid StudentId { get; set; }
        [Required]
        public string FullName { get; set; }
        [Required]
        public Guid CampusId { get; set; }
        [Required]
        public string CampusName { get; set; }
        [Required]
        public Guid ShiftId { get; set; }
        [Required]
        public string ShiftName { get; set; }
        [Required]
        public Guid ProgramDetailId { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public Guid SessionId { get; set; }
        [Required]
        public string SessionName { get; set; }

        [Required]
        public string ClassName { get; set; }


    }

    [Table("PreviousAcademic", Schema = "Admission")]
    public partial class PreviousAcademicRecord
    {
        [Key]
        [Required]
        public Guid PreviousAcademicId { get; set; }
        [Required]
        public Guid StudentId { get; set; }

        [Required]
        public Guid DegreeId { get; set; }
        [Column(TypeName = "jsonb")]
        public string AcademicMarksDetail { get; set; }
        [Required]
        public int StatusId { get; set; }

    }

    public partial class ProgramCourseList
    {
        [Key]
        [Required]
        public Guid CourseId { get; set; }
        [Required]
        public string FullName { get; set; }

    }

    [Table("StudentChallanAttachments", Schema = "Fee")]
    public partial class StudentChallanAttachments
    {
                [Key]
                [Required]
                public Guid StudentChallanAttachmentsId { get; set; }

                public Guid AdmissionFormId { get; set; }

                public Guid ClassId { get; set; }

                public string Image { get; set; }
                public string FileName { get; set; }

                public int InstallmentNo { get; set; }

                public int CreatedOn { get; set; }
                public int StatusId { get; set; }


    }



public class AdmissionFormRequest
{
    [Key]
    public Guid StudentChallanAttachmentId { get; set; }
    public Guid AdmissionFormId { get; set; }
    public Guid ClassId { get; set; }
    public int InstallmentNo { get; set; }
    public string Image { get; set; }
    public int StatusId { get; set; }
    public string FullName { get; set; }
        public string FileName { get; set; }

}

}

File: src/ui/app/root/application/models/Admission/Students.ts
export interface IAdmissionStudents {
	studentId: string;
	fullName: string;
	fatherName: string;
	studentCNIC: string;
	parentCNIC: string;
	studentContactNo: string;
	parentContactNo: string;
	guardians: string;
	genderId: string;
	dateOfBirth: Date;
	address: string;
	bloodGroupId: string;
	religionId: string;
	statusId: number;
	loggerId: string;
	academicInfo: string;
	image: string;
	operation:string;
}
export interface IAdmissionStudentsEx {
	studentId: string;
	fullName: string;
	fatherName: string;
	studentCNIC: string;
	parentCNIC: string;
	studentContactNo: string;
	parentContactNo: string;
	guardians: string;
	genderId: string;
	dateOfBirth: Date;
	address: string;
	bloodGroupId: string;
	religionId: string;
	statusId: number;
	loggerId: string;
	academicInfo: string;
	image: string;
	guardianName: string;
	degreeId: string;
	groupId: string;
	rollNo: string;
	obtainMarks: string;
	totalMarks: string;
	year: string;
	board: string;
	institute: string;
	registrationNo: string;


}
export interface ExamUpdList
{
	examDetailId: string;
	examMasterId: string;
	examTypeId: string;
	programCourseLinkId: string;
	total: number;
	obtained: number;
	examType: string;
	fullName: string;
	checked:boolean;
	admissionFormId:string;
	month:string;
	monthDate:string;
	examTypeName:string;
	attendanceStatusId:string;
	attendance:string;



}

export interface IAdmissionStudentsExx {
	campusProgramId: string;
	studentId: string;
	fullName: string;
	fatherName: string;
	studentCNIC: string;
	parentCNIC: string;
	studentContactNo: string;
	parentContactNo: string;
	guardians: string;
	genderId: string;
	dateOfBirth: Date;
	address: string;
	bloodGroupId: string;
	religionId: string;
	statusId: number;
	loggerId: string;
	academicInfo: string;
	image: string;
	guardianName: string;
	degreeId: string;
	groupId: string;
	rollNo: string;
	obtainMarks: string;
	totalMarks: string;
	year: string;
	board: string;
	institute: string;
	registrationNo: string;


}

export interface StudentChallanINfoData {
	challanNo: string;
	installmentNo: number;
	fullName: string;
	dueDate: string;
	paidDate?: string;
	feeAmount: number;
	concession: string;
}

export interface PreviousAcademicRecord {
	previousAcademicId: string;
	studentId: string;
	degreeId: string;
	academicMarksDetail: string;
	statusId: number;
}

export interface IAcademicMarks {
    newID: string;
    studentId: string;
    courseName: string;
    totalMarks: string;
    obtainedMarks: string;
    degree: string;
}
export interface ProgramCourseList {
	courseId: string;
	fullName: string;
}
export interface IAcademicInfoss {
    degreeId: string;
    groupId: string;
    institute: string;
    board: string;
    rollNo: string;
    year: string;
    obtainMarks: number;
    totalMarks: number;
}
export interface IAdmissionformList {


 
    subcity: string;
    programdetailid: string;
    genderid: string;
    year: string;
    percentage: number;
    dob: Date;
    fullname: string;
    fathername: string;
    studentcnic: string;
    parentcnic: string;
    studentcontactno: string;
    parentcontactno: string;
    guardian: string;
    address: string;
    academicinfo: string;
    boardregistrationno: string;
    email: string;
    flag: string;
    degreeid: string;
    mail: string;
    refferenceNo: string;


 
}export interface IAdmissionformListEx {


 
    subcity: string;
    programdetailid: string;
    genderid: string;
    year: string;
    percentage: number;
    dob: Date;
    fullname: string;
    fathername: string;
    studentcnic: string;
    parentcnic: string;
    studentcontactno: string;
    parentcontactno: string;
    guardian: string;
    address: string;
    academicinfo: string;
    boardregistrationno: string;
    email: string;
    flag: string;
    degreeid: string;
    mail: string;
    refferenceNo: string;
    challanNo:string;
	feeAmount:string;

 
}